openpype/modules/kitsu/kitsu_module.py

```diff
@@ -0,0 +1,477 @@
+    all_projects = gazu.project.all_projects()
+    bulk_writes = []
+    for project in all_projects:
+        # Create project locally
+        # Try to find project document
+        project_name = project["name"]
+        project_code = project_name
+        dbcon.Session["AVALON_PROJECT"] = project_name
+        project_doc = dbcon.find_one({"type": "project"})
+
+        print(f"Synchronizing {project_name}...")
+
+        # Get all assets from zou
+        all_assets = all_assets_for_project(project)
+        all_episodes = all_episodes_for_project(project)
+        all_seqs = all_sequences_for_project(project)
+        all_shots = all_shots_for_project(project)
+        all_entities = [
+            e
+            for e in all_assets + all_episodes + all_seqs + all_shots
+            if e["data"] and not e["data"].get("is_substitute")
+        ]
+
+        # Sync project. Create if doesn't exist
+        bulk_writes.append(sync_project(project, dbcon))
+
+        # Query all assets of the local project
+        project_col = dbcon.database[project_code]
+        zou_ids_and_asset_docs = {
+            asset_doc["data"]["zou"]["id"]: asset_doc
+            for asset_doc in project_col.find({"type": "asset"})
+            if asset_doc["data"].get("zou", {}).get("id")
+        }
+        zou_ids_and_asset_docs[project["id"]] = project_doc
+
+        # Create
+        to_insert = []
+        to_insert.extend(
+            [
+                create_op_asset(item)
+                for item in all_entities
+                if item["id"] not in zou_ids_and_asset_docs.keys()
+            ]
+        )
+        if to_insert:
+            # Insert doc in DB
+            project_col.insert_many(to_insert)
+
+            # Update existing docs
+            zou_ids_and_asset_docs.update(
+                {
+                    asset_doc["data"]["zou"]["id"]: asset_doc
+                    for asset_doc in project_col.find({"type": "asset"})
+                    if asset_doc["data"].get("zou")
+                }
+            )
+
+        # Update
+        bulk_writes.extend(
+            [
+                UpdateOne({"_id": id}, update)
+                for id, update in update_op_assets(all_entities, zou_ids_and_asset_docs)
+            ]
+        )
+
+        # Delete
+        diff_assets = set(zou_ids_and_asset_docs.keys()) - {
+            e["id"] for e in all_entities + [project]
+        }
+        if diff_assets:
+            bulk_writes.extend(
+                [
+                    DeleteOne(zou_ids_and_asset_docs[asset_id])
+                    for asset_id in diff_assets
+                ]
+            )
+
+        # Write into DB
+        if bulk_writes:
+            project_col.bulk_write(bulk_writes)
+
+    dbcon.uninstall()
+
+    # Run listening
+    if listen:
+        start_listeners()
+
+
+@cli_main.command()
+def listen():
+    """Listen to Kitsu server."""
+    start_listeners()
+
+
+@cli_main.command()
+def sign_in():
```

Comment:
I don't think this should be cli function but module should handle this on `tray_start` if user is not logged in.